### PR TITLE
Enable support for restarting application process if OpenGL context i…

### PR DESCRIPTION
**Pull request by rampr**
…s lost

Changes from -> https://github.com/axmolengine/axmol/pull/1409

Looks like other frameworks restart on context loss and cocos should do the same, A new event event_app_restarting is available for applications to listen to do their work if the state must be saved etc..

### CODE REVIEW

**Review by shijasmhd — APPROVED** [review]

File: cocos/platform/android/javaactivity-android.cpp [hunk 1]
```diff
@@ -115,6 +115,23 @@ JNIEXPORT void Java_org_cocos2dx_lib_Cocos2dxRenderer_nativeInit(JNIEnv*  env, j
     cocos2d::network::_preloadJavaDownloaderClass();
 }
 
+JNIEXPORT void JNICALL Java_org_axmol_lib_AxmolRenderer_nativeOnContextLost(JNIEnv*, jclass)
```
Comment (abhinavroy-hc) [inline cocos/platform/android/javaactivity-android.cpp]: @rampr shouldn't this be Java_org_cocos2dx_lib_Cocos2dxRenderer_nativeOnContextLost?

Comment (rampr) [reply cocos/platform/android/javaactivity-android.cpp]: you are right, will fix @abhinavroy-hc

Comment (rampr) [reply cocos/platform/android/javaactivity-android.cpp]: updated - Please check @abhinavroy-hc

Comment (abhinavroy-hc) [reply cocos/platform/android/javaactivity-android.cpp]: @rampr also Cocos2dxRenderer instead of AxmolRenderer

Comment (rampr) [reply cocos/platform/android/javaactivity-android.cpp]: Thanks @abhinavroy-hc. Fixed that also

**Review by abhinavroy-hc — APPROVED** [review]

**Review by thesayraj — APPROVED** [review]

**Review by hatim-developer — APPROVED** [review]

**Pull request merged**